src/Alert/Alert.tsx

```diff
@@ -0,0 +1,194 @@
+/** @jsx jsx */
+import { jsx, ClassNames } from "@emotion/core";
+import React, { CSSProperties, Fragment } from "react";
+import PropTypes from "prop-types";
+import classnames from "classnames";
+
+import { base } from "../typography";
+import { IconClose } from "../icons/IconClose";
+import { colors, PaletteColor } from "../colors";
+import { getOffsetInPalette } from "../colors/utils/getOffsetInPalette";
+
+interface AlertProps {
+  /**
+   * color theme for alert
+   * @default "light"
+   */
+  theme?: "light" | "dark";
+
+  /**
+   * The color used for the heading icon, the heading text will be 2 shades
+   * darker if light theme and 2 shaders lighter if a dark theme
+   */
+  color: PaletteColor;
+
+  /**
+   * The icon displayed to the left of the heading text
+   */
+  Icon: React.ComponentType<{ className?: string }>;
```

Comment:
I'd like to change this to be `React.ReactElement<{ className?: string }>`. 

If we pass an element, then we can render any element with whatever props we want and then decorate it with this component much the same way we decorate all the `as` props. This allows us to configure anything else we wanted on the element. If we go the route of using `ComponentType`, then we have to assume the only configurable prop we'd pass in is `className` or we'd have to do some trickery to allow us to pass other props in. For example, if we used a Space Kit icon and we wanted to use `weight="thin"`, that's not possible if we use `React.ComponentType`, but it's trivial if we use an element: `icon={<SomeIcon weight="thin" />}`.

<details>
<summary>Here's the patch to apply</summary>

I'm sure you already know this, but as a refresher, copy this including a trailing line break and use `pbpaste | git apply -p1` to apply this patch yourself.

```diff
diff --git a/src/Alert/Alert.story.mdx b/src/Alert/Alert.story.mdx
index 13cc113..f9714d9 100644
--- a/src/Alert/Alert.story.mdx
+++ b/src/Alert/Alert.story.mdx
@@ -31,7 +31,7 @@ Informational alerts present general info about something you are doing or worki
     <Alert
       onClose={() => undefined}
       color={colors.blue.base}
-      Icon={IconInfoSolid}
+      icon={<IconInfoSolid />}
       heading="Informational alert"
     >
       This is a general informational message telling you about something. Learn
@@ -42,7 +42,7 @@ Informational alerts present general info about something you are doing or worki
     <Alert
       onClose={() => undefined}
       color={colors.blue.base}
-      Icon={IconInfoSolid}
+      icon={<IconInfoSolid />}
       theme="dark"
       heading="Informational alert"
     >
@@ -59,7 +59,7 @@ Informational alerts present general info about something you are doing or worki
     <Alert
       onClose={() => undefined}
       color={colors.blue.base}
-      Icon={IconInfoSolid}
+      icon={<IconInfoSolid />}
       heading="Informational alert"
       extended={true}
     >
@@ -71,7 +71,7 @@ Informational alerts present general info about something you are doing or worki
     <Alert
       onClose={() => undefined}
       color={colors.blue.base}
-      Icon={IconInfoSolid}
+      icon={<IconInfoSolid />}
       theme="dark"
       heading="Informational alert"
       extended={true}
@@ -91,7 +91,7 @@ Warning alerts present information about the action that is about to be taken th
     <Alert
       onClose={() => undefined}
       color={colors.orange.base}
-      Icon={IconWarningSolid}
+      icon={<IconWarningSolid />}
       heading="Warning alert"
       actions={<Button size="small">Confirm</Button>}
     >
@@ -103,7 +103,7 @@ Warning alerts present information about the action that is about to be taken th
     <Alert
       onClose={() => undefined}
       color={colors.orange.base}
-      Icon={IconWarningSolid}
+      icon={<IconWarningSolid />}
       heading="Warning alert"
       theme="dark"
       actions={
@@ -127,7 +127,7 @@ Error alerts let the user know that something has gone wrong or is blocked in th
     <Alert
       onClose={() => undefined}
       color={colors.red.base}
-      Icon={IconErrorSolid}
+      icon={<IconErrorSolid />}
       heading="Error alert"
       style={{ marginBottom: 20 }}
     >
@@ -138,7 +138,7 @@ Error alerts let the user know that something has gone wrong or is blocked in th
     <Alert
       onClose={() => undefined}
       color={colors.red.base}
-      Icon={IconErrorSolid}
+      icon={<IconErrorSolid />}
       heading="Error alert"
       theme="dark"
       style={{ marginBottom: 20 }}
@@ -157,7 +157,7 @@ Success alerts are confirmation that the action the user was trying to take has
     <Alert
       onClose={() => undefined}
       color={colors.green.base}
-      Icon={IconSuccessSolid}
+      icon={<IconSuccessSolid />}
       heading="Success alert"
     >
       A new schema registry has been created by timbotnik.
@@ -167,7 +167,7 @@ Success alerts are confirmation that the action the user was trying to take has
     <Alert
       onClose={() => undefined}
       color={colors.green.base}
-      Icon={IconSuccessSolid}
+      icon={<IconSuccessSolid />}
       theme="dark"
       heading="Success alert"
     >
diff --git a/src/Alert/Alert.tsx b/src/Alert/Alert.tsx
index d22dd52..ab5c1d1 100644
--- a/src/Alert/Alert.tsx
+++ b/src/Alert/Alert.tsx
@@ -25,7 +25,7 @@ interface AlertProps {
   /**
    * The icon displayed to the left of the heading text
    */
-  Icon: React.ComponentType<{ className?: string }>;
+  icon: React.ReactElement<{ className?: string }>;
 
   heading: React.ReactNode;
 
@@ -73,7 +73,7 @@ export const Alert: React.FC<AlertProps> = ({
   headingAs = "h2",
   children,
   color,
-  Icon,
+  icon,
   theme = "light",
   extended = false,
   ...otherProps
@@ -120,9 +120,23 @@ export const Alert: React.FC<AlertProps> = ({
               ),
               children: (
                 <Fragment>
-                  <Icon
-                    css={{ width: 20, height: 20, color, marginRight: 13 }}
-                  />
+                  <ClassNames>
+                    {({ css, cx }) =>
+                      React.cloneElement(icon, {
+                        className: classnames(
+                          icon.props.className,
+                          cx(
+                            css({
+                              width: 20,
+                              height: 20,
+                              color,
+                              marginRight: 13,
+                            })
+                          )
+                        ),
+                      })
+                    }
+                  </ClassNames>
                   {heading}
                 </Fragment>
               ),
@@ -190,5 +204,5 @@ Alert.propTypes = {
       .map((color) => Object.values(color))
       .reduce((a, b) => a.concat(b)) as PaletteColor[]
   ).isRequired,
-  Icon: PropTypes.func.isRequired,
+  icon: PropTypes.element.isRequired,
 };
```
</details>

Reply:
yep that makes a lot of sense! applied the patch 👍 
I've never actually encounter a patch before, very handy!